scripts/install/install-dependencies.sh

```diff
@@ -0,0 +1,394 @@
+#!/bin/bash -e
+
+THIS_DIR="$(dirname $(readlink -f "${0}"))"
+THIS_PROGRAM="$(basename "${0}")"
+
+if [[ -z "${PANDIR}" || -z "${POCS}" || -z "${PAWS}" || -z "${PANLOG}" ||
+      -z "${PANUSER}" || ! -d "${PANDIR}" ]] ; then
+  echo "Please set the Panoptes environment variables, then re-run this script."
+  exit 1
+fi
+
+cd "${PANDIR}"
+
+# TODO(jamessynge): Add flags to control behavior, such as skipping apt-get.
+
+ASTROMETRY_VERSION="0.72"
+INSTALL_PREFIX="/usr/local"
+DO_APT_GET=1
+DO_MONGODB=1
+DO_CONDA=1
+DO_CFITSIO=0
+DO_ASTROMETRY=1
+DO_ASTROMETRY_INDICES=1
+DO_ASTROMETRY_FULL_INDICES=0
+DO_PIP_REQUIREMENTS=1
+
+function echo_bar() {
+  eval $(resize|grep COLUMNS=)
+  printf "%${COLUMNS:-80}s\n" | tr ' ' '#'
+}
+
+# Append $1 to PATH and write command to do the same to .bashrc.
+function add_to_PATH() {
+  local -r the_dir="$(readlink -f "${1}")"
+  if [[ -z "$(egrep -- "PATH=.*${the_dir}" ~/.bashrc)" ]] ; then
+    echo >>"${HOME}/.bashrc" "
+# Added by PANOPTES install-dependencies.sh
+PATH=\"${the_dir}:\${PATH}\""
+    PATH="${the_dir}:${PATH}"
+    echo
+    echo "Added ${the_dir} to PATH in .bashrc"
+  else
+    echo ".bashrc already adds ${the_dir} to PATH"
+  fi
+}
+
+# Given the path to a pkg-config file (.pc), extract the version number.
+function extract_version_from_pkg_config() {
+  if [[ -f "${1}" ]] ; then
+    egrep '^Version:' "${1}" | cut '-d:' -f2
+  else
+    echo ""
+  fi
+}
+
+function install_apt_packages() {
+  # Remove all the comments from the package list and install the packages whose
+  # names are left.
+  APT_PKGS="$(cut '-d#' -f1 "${THIS_DIR}/apt-packages-list.txt" | sort | uniq)"
+  echo
+  echo "Running sudo apt-get install, you may be prompted for your password."
+  echo
+  (set -x ; sudo apt-get update)
+  (set -x ; sudo apt-get install --yes ${APT_PKGS})
+}
+
+function install_mongodb() {
+  # This is based on https://www.howtoforge.com/tutorial/install-mongodb-on-ubuntu-16.04/
+  # Note this function does not configure mongodb itself, i.e. no users or
+  # security settings.
+  echo_bar
+  echo "
+Installing mongodb, for which several commands require sudo, so
+you may be prompted for you password.
+"
+  sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv EA312927
+  echo "deb http://repo.mongodb.org/apt/ubuntu "$(lsb_release -sc)"/mongodb-org/3.2 multiverse" \
+     | sudo tee /etc/apt/sources.list.d/mongodb-org-3.2.list
+  sudo apt-get update
+  sudo apt-get install -y mongodb-org
+  echo "[Unit]
+Description=High-performance, schema-free document-oriented database
+After=network.target
+Documentation=https://docs.mongodb.org/manual
+
+[Service]
+User=mongodb
+Group=mongodb
+ExecStart=/usr/bin/mongod --quiet --config /etc/mongod.conf
```

Comment:
We should also specify a `--dbpath` (or alter the `mongod.conf`). I've put this in `$PANDIR/data` previously. Our current setup writes a lot of entries to the database and it grows very large (usually needs > 4GB to run). On PAN001 all of $PANDIR is on a separate mount from the rest of `/var` (and everything else)

Reply:
Note also that mongo does not require you to actually create a collection (read: database) before using it. You can just start writing to something and it will store properly

Reply:
Not surprisingly, I didn't know that info about --dbpath.
Does mongodb allow multiple dbpath directories (i.e. can we arrange for our collections to be in /var/panoptes)?

No change made yet.

Reply:
There is some info (for instance, [here](https://docs.mongodb.com/manual/administration/configuration/#run-multiple-database-instances-on-the-same-system)) but I think the better answer would be to run what we want in a container.